diesel_cli/src/infer_schema_internals/sqlite.rs

```diff
@@ -176,6 +176,67 @@ impl QueryableByName<Sqlite> for PrimaryKeyInformation {
     }
 }
 
+#[cfg(feature = "integer_primary_key_i64_for_sqlite_3_37")]
+struct WithoutRowIdInformation {
+    name: String,
+    without_row_id: bool,
+}
+
+#[cfg(feature = "integer_primary_key_i64_for_sqlite_3_37")]
+impl QueryableByName<Sqlite> for WithoutRowIdInformation {
+    fn build<'a>(row: &impl NamedRow<'a, Sqlite>) -> deserialize::Result<Self> {
+        Ok(Self {
+            name: NamedRow::get::<Text, String>(row, "name")?,
+            without_row_id: NamedRow::get::<Bool, bool>(row, "wr")?,
+        })
+    }
+}
+
+#[cfg(feature = "integer_primary_key_i64_for_sqlite_3_37")]
+pub fn column_is_row_id(
+    conn: &mut SqliteConnection,
+    table: &TableName,
+    column_name: &str,
+    type_name: &str,
+) -> QueryResult<bool> {
+    let sqlite_version = get_sqlite_version(conn)?;
+    if sqlite_version < SqliteVersion::new(3, 37, 0) {
+        return Ok(false);
+    }
+
+    if type_name != "integer" {
+        return Ok(false);
+    }
+
+    let table_xinfo_query = format!("PRAGMA TABLE_XINFO('{}')", &table.sql_name);
+    let table_xinfo_results = sql_query(table_xinfo_query).load::<PrimaryKeyInformation>(conn)?;
+
+    let primary_keys = table_xinfo_results
+        .iter()
+        .filter(|pk_info| pk_info.primary_key)
+        .collect::<Vec<_>>();
+
+    if primary_keys.len() != 1 {
+        return Ok(false);
+    }
+
+    if primary_keys[0].name != column_name {
+        return Ok(false);
+    }
+
+    let table_list_query = format!("PRAGMA TABLE_LIST('{}')", &table.sql_name);
+    let table_list_results = sql_query(table_list_query).load::<WithoutRowIdInformation>(conn)?;
+
+    match table_list_results
+        .iter()
+        .find(|wr_info| wr_info.name == table.sql_name)
+        .map(|wr_info| wr_info.without_row_id)
+    {
+        Some(false) => Ok(true),
+        _ => Ok(false),
+    }
```

Comment:
```suggestion
    let res = table_list_results
        .iter()
        .find(|wr_info| wr_info.name == table.sql_name)
        .map(|wr_info| !wr_info.without_row_id)
        .unwrap_or_default();
    Ok(res)
```

looks a bit cleaner.